vividus-bdd-engine/src/test/java/org/vividus/bdd/steps/CollectionComparisonRuleTests.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.steps;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.DynamicTest.dynamicTest;
+import static org.vividus.bdd.steps.CollectionComparisonRule.ARE_EQUAL_TO;
+import static org.vividus.bdd.steps.CollectionComparisonRule.ARE_EQUAL_TO_ORDERED_COLLECTION;
+import static org.vividus.bdd.steps.CollectionComparisonRule.CONTAIN;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+
+class CollectionComparisonRuleTests
+{
+    @TestFactory
+    Stream<DynamicTest> listRules()
```

Comment:
From my point of view, it is easier to read using a data source

CONTAIN, List.of(1, 0), List.of(1, 1, 1), false
where
rule, expected, actual, comparisonResult

Reply:
depends on :)